lib/active_admin/middleware/error_handler.rb

```diff
@@ -0,0 +1,54 @@
+module ActiveAdmin
+  class ErrorHandler
+
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      @env = env
+      @app.call(env)
+    rescue Exception => exception
+      if is_active_admin_error? && use_active_admin_error_page?
+        render_exception(exception)
+      else
+        raise exception
+      end
+    end
+
+    private
+
+    def is_active_admin_error?
+      stringified_namespaces.include?(current_namespace)
+    end
+
+    def use_active_admin_error_page?
+      if ActiveAdmin.application.use_active_admin_error_page.is_a?(Proc)
+        ActiveAdmin.application.use_active_admin_error_page.call
+      else
+        ActiveAdmin.application.use_active_admin_error_page
+      end
```

Comment:
We can remove `use_active_admin_error_page?` and add the middleware only if ActiveAdmin.application.use_active_admin_error_page or ActiveAdmin.application.use_active_admin_error_page.call is true